src/commands/run.rs

```diff
@@ -13,24 +25,121 @@ impl Run {
+                    std::io::ErrorKind::Other,
+                    format!("failed to get current binary path: {}", e),
+                ));
+            }
+        };
+
+        if serve {
+            debug!(
+                "Setting up aardvark server with input directory as {:?}",
+                input_dir
+            );
+
+            println!("current binary path{:?}", std::env::current_exe());
+
+            if let Err(er) = serve::serve(&input_dir, port, &filter_search_domain) {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("Error starting server {}", er),
+                ));
+            }
+            Ok(())
+        } else {
+            // double fork and verify if server is running
+            // and exit parent
+            // Why double fork ?
+            // Its important that nature of aardvark server is more like a daemon
+            // so following block ensures that process returns when aardvark server
+            // is ready to serve
+            //
+            // setsid() ensures that there is no controlling terminal on the child process
+
+            match unsafe { fork() } {
+                Ok(ForkResult::Parent { child, .. }) => {
+                    log::debug!("starting aardvark on a child with pid {}", child);
+                    if let Err(err) = waitpid(Some(child), None) {
+                        log::debug!("error while waiting for child pid {}", err);
+                    }
+                    // verify aardvark here
+                    Ok(())
+                }
+                Ok(ForkResult::Child) => {
+                    let mut aardvark_args = vec![];
+                    let port_cloned = port.clone().to_string();
+                    // only use systemd when it is booted, see sd_booted(3)
+                    if Path::new(SYSTEMD_CHECK_PATH).exists()
+                        && Run::is_executable_in_path(SYSTEMD_RUN)
+                    {
+                        aardvark_args = vec![SYSTEMD_RUN, "-q", "--scope"];
+
+                        if rootless {
+                            aardvark_args.push("--user");
+                        }
+                    }
+
+                    aardvark_args.extend(vec![
+                        current_binary_path.as_str(),
+                        "--config",
+                        &input_dir,
+                        "-p",
+                        port_cloned.as_str(),
+                        "--serve",
+                        "true",
+                        "run",
+                    ]);
+
+                    log::debug!("start aardvark-dns: {:?}", aardvark_args);
+
+                    let output = match log_enabled!(log::Level::Debug) {
+                        true => Stdio::inherit(),
+                        false => Stdio::null(),
+                    };
+
+                    Command::new(&aardvark_args[0])
```

Comment:
do not exec, we are already aardvark just continue with our main logic

Reply:
Yeah instead of exec we can just run the server but the process would be running without systemd-run ?

Reply:
netavark should still launch it with systemd-run